core/src/main/java/com/linecorp/armeria/common/QueryParamsBuilder.java

```diff
@@ -0,0 +1,523 @@
+     *         no such value or it can't be converted into {@code float}.
+     */
+    @Nullable
+    Float getFloatAndRemove(String name);
+
+    /**
+     * Removes all the parameters with the specified name and returns the parameter value which was added
+     * first.
+     *
+     * @param name the parameter name
+     * @param defaultValue the default value
+     * @return the {@code float} value of the first value in insertion order or {@code defaultValue} if there
+     *         is no such value or it can't be converted into {@code float}.
+     */
+    float getFloatAndRemove(String name, float defaultValue);
+
+    /**
+     * Removes all the parameters with the specified name and returns the parameter value which was added
+     * first.
+     *
+     * @param name the parameter name
+     * @return the {@code double} value of the first value in insertion order or {@code null} if there is
+     *         no such value or it can't be converted into {@code double}.
+     */
+    @Nullable
+    Double getDoubleAndRemove(String name);
+
+    /**
+     * Removes all the parameters with the specified name and returns the parameter value which was added
+     * first.
+     *
+     * @param name the parameter name
+     * @param defaultValue the default value
+     * @return the {@code double} value of the first value in insertion order or {@code defaultValue} if there
+     *         is no such value or it can't be converted into {@code double}.
+     */
+    double getDoubleAndRemove(String name, double defaultValue);
+
+    /**
+     * Removes all the parameters with the specified name and returns the parameter value which was added
+     * first.
+     *
+     * @param name the parameter name
+     * @return the milliseconds value of the first value in insertion order or {@code null} if there is no such
+     *         value or it can't be converted into milliseconds.
+     */
+    @Nullable
+    Long getTimeMillisAndRemove(String name);
+
+    /**
+     * Removes all the parameters with the specified name and returns the parameter value which was added
+     * first.
+     *
+     * @param name the parameter name
+     * @param defaultValue the default value
+     * @return the milliseconds value of the first value in insertion order or {@code defaultValue} if there is
+     *         no such value or it can't be converted into milliseconds.
+     */
+    long getTimeMillisAndRemove(String name, long defaultValue);
+
+    /**
+     * Adds a new parameter with the specified {@code name} and {@code value}.
+     *
+     * @param name the parameter name
+     * @param value the parameter value
+     * @return {@code this}
+     */
+    QueryParamsBuilder add(String name, String value);
+
+    /**
+     * Adds new parameters with the specified {@code name} and {@code values}. This method is semantically
+     * equivalent to
+     * <pre>{@code
+     * for (String value : values) {
+     *     params.add(name, value);
```

Comment:
Let me rename it to `builder`. `paramBuilder` sounds fine, but `builder` is shorter and I guess users already know it's about `QueryParamsBuilder` (because they are browsing its Javadoc).